experiments/gym/train_lunar_lander.py

```diff
@@ -80,108 +86,241 @@ def forward(self, obs):
+        env (DictGymWrapper): the Gym wrapper
+        memory_proxy (TableMemoryProxy): the proxy for the memory
+        data_group (str): the data_group to receive data batches
+    Returns:
+        (list[Callback]): the callbacks for the SAC RL training
+    """
+    device = torch.device(args.device)
+    batch_size = args.batch_size
+    max_grad_norm = 1
+    len_rollout = args.rollout_length
+    num_actions = env.dict_space.actions.shape[0]
 
-    logged_cbs = [
+    init_alpha = 1.0
+    ln_alpha = torch.tensor(np.log(init_alpha), requires_grad=True, device=device)
+    training_cbs = [
         QLoss(
             name="q1",
             q=q1,
-            opt=Adam(q1.parameters(), lr=learning_rate),
+            opt=Adam(q1.parameters(), lr=args.critic_lr),
             max_grad_norm=max_grad_norm,
+            data_group=data_group,
         ),
         QLoss(
             name="q2",
             q=q2,
-            opt=Adam(q2.parameters(), lr=learning_rate),
+            opt=Adam(q2.parameters(), lr=args.critic_lr),
             max_grad_norm=max_grad_norm,
+            data_group=data_group,
         ),
         PolicyLoss(
-            pi=policy,
+            pi=policy_proxy.policy,
             ln_alpha=ln_alpha,
             q=q1,
-            opt=Adam(policy.parameters(), lr=learning_rate),
+            opt=Adam(policy_proxy.policy.parameters(), lr=args.actor_lr),
             max_grad_norm=max_grad_norm,
+            data_group=data_group,
         ),
         AlphaLoss(
-            pi=policy,
+            pi=policy_proxy.policy,
             ln_alpha=ln_alpha,
-            opt=Adam([ln_alpha], lr=learning_rate),
+            opt=Adam([ln_alpha], lr=args.actor_lr),
             n_actions=num_actions,
             max_grad_norm=max_grad_norm,
             max_alpha=10.0,
+            data_group=data_group,
         ),
         QTarget(
-            pi=policy,
+            pi=policy_proxy.policy,
             ln_alpha=ln_alpha,
             q1=q1,
             q2=q2,
-            roll_length=rollout_len,
+            roll_length=len_rollout,
             reward_scale=0.1,
+            data_group=data_group,
         ),
         ThreadedGymCollector(
             env,
-            agent_proxy,
+            policy_proxy,
             memory_proxy,
             warmup_steps=batch_size,
             render=False,
         ),
     ]
+    return training_cbs
+
 
-    callbacks = logged_cbs + [
+def create_complementary_callbacks(
+    args,
+    train_cbs: list[Callback],
+):
+    """The function creates the supplementary callbacks for the training and adds them to the callback lists
+    and returns the list.
+
+        Arguments:
+            args:
+            train_cbs (list[Callback]):
+        Returns:
+            (list[Callback]): the full list of callbacks for the training
+    """
+    callbacks = train_cbs + [
         TensorboardLogger(
-            logged_cbs,
+            train_cbs,
```

Comment:
This seems wrong. The logger should only take logger cbs (things that inherit `LoggingMixin`)

Reply:
Sorry didn't get what is wrong here. 
All callbacks in train_cbs (LossCallback, BatchCallback, GymCollector) inherit LoggingMixin and Callback.